File: C2109L/ADF1/20-09-2022/javade02/untitled/src/main/java/com/aptech/de02/SavingsAccount.java
package com.aptech.de02;

public class SavingsAccount {
    private static float annualInterestRate;
    private Double savingsBalance;

    public SavingsAccount(Double savingsBalance) {
        this.savingsBalance = savingsBalance;
    }

    public Double calculateMonthlyInterest() {
        double interest = (annualInterestRate * savingsBalance)/12;
        this.savingsBalance += interest;
        return interest;
    }
    public static void modifyAnnualInterestRate(float newValue) {
        annualInterestRate = newValue;
    }

    @Override
    public String toString() {
        return "SavingsAccount{" +
                "savingsBalance=" + savingsBalance +
                '}';
    }
}

File: C2111L/de13/SavingsAccount.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace de13
{
    internal class SavingsAccount
    {
        private static double annualInterestRate;
        private double savingsBalance;
        public double CalculateMonthlyInterest() { 
            return savingsBalance * (annualInterestRate / 12);
        }
        public static void ModifyInterestRate(double newRate) { 
            annualInterestRate = newRate;
        }
        public double SavingBalance { get => savingsBalance; set => savingsBalance = value; }
        public override string ToString()
        {
            return $"Saving balance: {SavingBalance}, monthly interest: {CalculateMonthlyInterest()}";
        }
    }
}
